Updated Prometheus page with details of custom domains

kentsanggds opened the pull request:
## What 

Give users some guidance on what they need to do in order to get custom domains working with Prometheus.

Also provide some information about binding to just the first route and the behaviour if there are no routes bound to an app.

kentsanggds committed:
Updated Prometheus page with details of custom domains

philandstuff commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,23 @@ You can create a Prometheus service within your PaaS space and allow it to bind
   - `cf bind-service <app-name> <service-instance-name>`
+3. If your app uses a custom domain, you need to make sure that the [authorization header][] is being forwarded otherwise the basic auth to the metrics endpoint will fail.
+  - `cf update-service <cdn-route-service> \
+    -c '{"domain": "<custom-domain>", "headers": ["Accept", "Authorization"]}'`
```
maybe make clear that if the user has any other headers that they need to forward, they need to be listed here too?

kentsanggds replied:
ok

philandstuff commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,23 @@ You can create a Prometheus service within your PaaS space and allow it to bind
+* Only the first route of the app will be bound, any changes to the first route will be picked up within 5 minutes.
+* If there are no routes to your app the Prometheus service will default the route to `<app-name>.cloudapps.digital`
+
+
```
can we not have 4 blank lines in a row?

kentsanggds replied:
ok

philandstuff commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,23 @@ You can create a Prometheus service within your PaaS space and allow it to bind
 
+### Notes about the app route
+
+* Only the first route of the app will be bound, any changes to the first route will be picked up within 5 minutes.
```
can we say "scraped" not "bound" here? I feel this is closer to the language that users would understand and would use themselves

philandstuff replied:
Also, this is two separate sentences separated with a comma - can you use a full stop

kentsanggds replied:
Sure

philandstuff commented:
Also, can you book a slot with @soupdragon99 just to check this over?

kentsanggds committed:
Update to the monitor-paas-app-with-prometheus

Updated the html to make it clearer for users, and added information about how the broker handles routes.

philandstuff commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,21 @@ You can create a Prometheus service within your PaaS space and allow it to bind
   - `cf bind-service <app-name> <service-instance-name>`
+3. If your app uses a custom domain, you need to make sure that the [authorization header and other app headers][] are being forwarded otherwise the basic auth to the metrics endpoint will fail and your app won't have access to your other headers.
+  - `cf update-service <cdn-route-service> \
+    -c '{"domain": "<custom-domain>", "headers": ["Accept", "Authorization"]}'`
```
I guess my worry is that a user will simply copy/paste this command line, and might miss that they also need to forward, say, "Cookie" in their particular app's case.  I don't think the words "and other app headers" in the copy above are enough to prevent this.

Maybe:

 - `cf update-service <cdn-route-service> \
    -c '{"domain": "<custom-domain>", "headers": ["Accept", "Authorization", <any-other-headers-needed-by-your-app>]}'`

This has the benefit that the command line can't be simply copy-pasted; the user has to make an active choice to remove the "any other headers" bit.

kentsanggds replied:
👍

philandstuff commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,21 @@ You can create a Prometheus service within your PaaS space and allow it to bind
+### Notes about the app route
+
+* Only the first route of the app will be scraped.
+* Any changes to the first route will be picked up within 5 minutes.
```
Might be worth being explicit about zero-downtime deploys that change app routes. Something like:

- This means that if you use a zero-downtime plugin such as [autopilot][], prometheus may not immediately discover changes to your app route.

[autopilot]: https://github.com/contraband/autopilot

kentsanggds replied:
ok

kentsanggds committed:
Updated the doc for monitoring paas apps

- made the header values passed cf command clearer
- added information about the gaps in metrics when the first route of an app changes.

philandstuff reviewed (approved):
looks good to me - @soupdragon99 can you give a quick review too?

soupdragon99 commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,24 @@ You can create a Prometheus service within your PaaS space and allow it to bind
   - `cf create-service gds-prometheus prometheus <service-instance-name>`
 2. Either [update your app's manifest.yml] to bind your new service or bind it using the CLI
   - `cf bind-service <app-name> <service-instance-name>`
+3. If your app uses a custom domain, you need to make sure that the [authorization header and other app headers][] are being forwarded otherwise the basic auth to the metrics endpoint will fail and your app won't have access to your other headers.
```
'are being forwarded' where are they being forwarded to, the custom domain?

kentsanggds replied:
Forward to the app, which is located at the URL given in PaaS

soupdragon99 commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,24 @@ You can create a Prometheus service within your PaaS space and allow it to bind
-Within 5 minutes Prometheus will start scraping your application for metrics, you can validate this by checking [Grafana](https://grafana-paas.cloudapps.digital).
+Within 5 minutes Prometheus will start scraping your application for metrics, you can validate this by checking [Grafana][].
 
+### Notes about the app route
```
Can we change this from a heading to a lead-in line for the bullets below, I think this would work better. Do the bullets apply to just the custom domain step or all three steps?

kentsanggds replied:
OK, so drop the heading and just have it as a lead-in line.
The bullets apply to all domains, regardless if they are custom or not.

soupdragon99 commented on source/manuals/monitor-paas-app-with-prometheus.html.md.erb:
```diff
@@ -33,12 +33,24 @@ You can create a Prometheus service within your PaaS space and allow it to bind
 
+### Notes about the app route
+
+* Only the first route of the app will be scraped.
```
How common is it for apps to have multiple routes? Can we give an example of one that has multiple routes and one that doesn't? It would give some more context.

kentsanggds replied:
OK, so drop the heading and just have it as a lead-in line, the bullets apply to all domains, regardless if they are custom or not.

kentsanggds replied:
There is a possibility that apps will have multiple routes especially if they are using custom domains, so they might have something like this - 

`custom.domain.gov.uk, an-interesting-app.cloudapps.digital`, so only the first route will be picked up. 

I've got no idea how common it would be for apps to have this, but the behaviour of the service broker is such that it will ignore any routes after the first one so it's probably a good idea to let a user know in case they do have multiple routes.

soupdragon99 reviewed (changes requested):
I have made some suggestions, happy to have a chat about them if that's easier.

soupdragon99 committed:
tech writer edits

soupdragon99 committed:
Update monitor-paas-app-with-prometheus.html.md.erb

kentsanggds commented:
@soupdragon99 I think the updates look good, if you're happy with them can you approve and then I can merge
